app/controllers/course/announcements_controller.rb

```diff
@@ -4,6 +4,10 @@ class Course::AnnouncementsController < Course::ModuleController
   def index #:nodoc:
     @announcements = @announcements.includes(:creator).sorted_by_sticky.sorted_by_date
     @announcements = @announcements.page(params[:page])
+    # mark each announcement in this page as read
+    @announcements.each do |ann|
```

Comment:
Alright...Let me think...also thanks for your comments. I will fix those problems.